src/python/grpcio_observability/README.rst

```diff
@@ -1,2 +1,91 @@
-### TODO(xuanwn)
```

Comment:
I think it'd be a good idea to also publish API documentation [like this](https://github.com/grpc/grpc/blob/master/doc/python/sphinx/grpc_health_checking.rst).